src/Components/Browser.JS/src/Services/UriHelper.ts

```diff
@@ -1,69 +1,88 @@
 import '@dotnet/jsinterop';
 
-let hasRegisteredEventListeners = false;
+let hasRegisteredNavigationInterception = false;
+let hasRegisteredNavigationEventListeners = false;
 
 // Will be initialized once someone registers
 let notifyLocationChangedCallback: { assemblyName: string; functionName: string } | null = null;
 
 // These are the functions we're making available for invocation from .NET
 export const internalFunctions = {
+  listenForNavigationEvents,
   enableNavigationInterception,
   navigateTo,
   getBaseURI: () => document.baseURI,
   getLocationHref: () => location.href,
 };
 
-function enableNavigationInterception(assemblyName: string, functionName: string) {
-  if (hasRegisteredEventListeners || assemblyName === undefined || functionName === undefined) {
+function listenForNavigationEvents(assemblyName: string, functionName: string) {
+  if (hasRegisteredNavigationEventListeners || assemblyName === undefined || functionName === undefined) {
```

Comment:
I just had a look and can't find any scenario where `assemblyName === undefined || functionName === undefined`. Unless you specifically know how those could happen, could we remove those clauses?

I know they were already there before your change, but maybe they can be cleaned away if they became redundant at some point in the past.